jung-visualization/src/main/java/edu/uci/ics/jung/visualization/subLayout/TreeCollapser.java

```diff
@@ -49,6 +52,29 @@ public static void expand(MutableCTreeNetwork tree, CTreeNetwork subTree) {
             ? tree.inEdges(subTree).iterator().next() // THERE CAN BE ONLY ONE
             : null;
     tree.removeNode(subTree);
-    TreeUtils.addSubTree(tree, subTree, parent.orElse(null), parentEdge);
+    if (parent.isPresent()) {
+      TreeUtils.addSubTree(tree, subTree, parent.get(), parentEdge);
+    } else {
+      // tree is empty, so just copy all of subTree into tree
+      copyInto(tree, subTree);
+    }
+  }
+
+  private static <N, E> void copyInto(MutableCTreeNetwork<N, E> to, CTreeNetwork<N, E> from) {
+    from.root()
```

Comment:
A couple of notes about this:

(1) IMO this syntax is not as clear as this would be:
    
    if (!from.root().isPresent()) {
      return;
    }
    root = from.root().get();
    to.addNode(root);
    ...

This version is also much less indented.
(That is, I think that this is one of those cases in which the fluent API is not doing us any favors.)

(2) Although we may not need this at all (see above), if we do want to keep this method we should consider putting it in graph.util.Graphs (or perhaps Trees).

Reply:
Thanks for the feedback! I'll act upon these suggestions if I end up keeping `copyInto`. :)